--- configure.in~	2001-04-24 22:25:39.000000000 +0300
+++ configure.in	2004-12-04 20:04:32.859154887 +0200
@@ -104,18 +104,6 @@
     if test x$ac_cv_prog_gcc = xyes; then
         CFLAGS="$CFLAGS -fexpensive-optimizations -fomit-frame-pointer"
     fi
-    case "$target" in
-        i486-*-*)  # Yeah right. :)
-            if test x$ac_cv_prog_gcc = xyes; then
-                CFLAGS="$CFLAGS -march=486"
-            fi
-            ;;
-        i?86-*-*)
-            if test x$ac_cv_prog_gcc = xyes; then
-                CFLAGS="$CFLAGS -march=pentium -mcpu=pentiumpro"
-            fi
-            ;;
-    esac
 fi
 
 dnl Check for SDL
