Index: smpeg.cpp
===================================================================
RCS file: /cvs/cvsroot/smpeg/smpeg.cpp,v
retrieving revision 1.20
retrieving revision 1.21
diff -u -r1.20 -r1.21
--- smpeg.cpp	4 Apr 2001 21:58:18 -0000	1.20
+++ smpeg.cpp	31 Dec 2003 04:48:47 -0000	1.21
@@ -32,6 +32,37 @@
     MPEG *obj;
 };
 
+
+static inline int sanityCheckByteorder(void)
+{
+    int sane = 0;
+    union {
+        Uint32 ui32;
+        Uint8 ui8[4];
+    } swapper;
+
+    swapper.ui32 = 0x00000001;
+
+#if SDL_BYTEORDER == SDL_LIL_ENDIAN
+    sane = (swapper.ui8[0] == 0x01);
+#else
+    sane = (swapper.ui8[3] == 0x01);
+#endif
+
+    if (!sane) {
+        fprintf(stderr,
+          "\n\n"
+          "*************************************************************\n"
+          " SMPEG ERROR: SDL is wrong about this platform's byte order!\n"
+          "   You need to fix your SDL install before SMPEG can work!\n"
+          "*************************************************************\n"
+          "\n\n");
+    }
+
+    return(sane);
+}
+
+
 /* Create a new SMPEG object from an MPEG file.
    On return, if 'info' is not NULL, it will be filled with information 
    about the MPEG object.
@@ -45,6 +76,9 @@
 {
     SMPEG *mpeg;
 
+    if (!sanityCheckByteorder())
+        return(NULL);
+
     /* Create a new SMPEG object! */
     mpeg = new SMPEG;
     mpeg->obj = new MPEG(file, sdl_audio ? true : false);
@@ -61,6 +95,9 @@
 {
     SMPEG *mpeg;
 
+    if (!sanityCheckByteorder())
+        return(NULL);
+
     /* Create a new SMPEG object! */
     mpeg = new SMPEG;
     mpeg->obj = new MPEG(file, sdl_audio ? true : false);
@@ -81,6 +118,9 @@
 {
     SMPEG *mpeg;
 
+    if (!sanityCheckByteorder())
+        return(NULL);
+
     /* Create a new SMPEG object! */
     mpeg = new SMPEG;
     mpeg->obj = new MPEG(data, size, sdl_audio ? true : false);
@@ -96,6 +136,9 @@
 {
     SMPEG *mpeg;
 
+    if (!sanityCheckByteorder())
+        return(NULL);
+
     /* Create a new SMPEG object! */
     mpeg = new SMPEG;
     mpeg->obj = new MPEG(src, sdl_audio ? true : false);
