Index: video/MPEGvideo.cpp
===================================================================
RCS file: /cvs/cvsroot/smpeg/video/MPEGvideo.cpp,v
retrieving revision 1.40
retrieving revision 1.41
diff -u -r1.40 -r1.41
--- video/MPEGvideo.cpp	18 Jun 2001 20:59:57 -0000	1.40
+++ video/MPEGvideo.cpp	31 Dec 2003 05:09:34 -0000	1.41
@@ -438,6 +438,10 @@
       SDL_FreeYUVOverlay(_image);
     }
     _image = SDL_CreateYUVOverlay(_srcrect.w, _srcrect.h, SDL_YV12_OVERLAY, dst);
+    if ( _image == NULL ) {
+        return false;
+    }
+
     if ( !_dstrect.w || !_dstrect.h ) {
         _dstrect.w = dst->w;
         _dstrect.h = dst->h;
@@ -502,6 +506,7 @@
     {
       SDL_FreeYUVOverlay(_image);
       _image = SDL_CreateYUVOverlay(_srcrect.w, _srcrect.h, SDL_YV12_OVERLAY, _dst);
+      /* !!! FIXME: Uhh...what if this one fails? */
     }
 
     SDL_mutexV( _mutex );
